hax/helper/generate_sysconf.py

```diff
@@ -298,22 +301,24 @@ def get_service_port(self, ep: str) -> int:
 
     def prepare_svc(self, svc_id: str, name: str):
         ep = self.provider.get_service_ep(svc_id)
+        if not ep:
+            raise RuntimeError('Cannot get service endpoint.')
         addr = self.get_service_addr(ep)
         port = self.get_service_port(ep)
 
-        checks = {}
+        checks: Dict[str, Any] = {}
         checks['args'] = ['/opt/seagate/cortx/hare/libexec/check-service']
         checks['interval'] = '1s'
         checks['status'] = 'warning'
         # get svc checks args as per svc name
         if name == 'hax':
             checks['args'].append('--hax')
         elif name in ('confd', 'ios'):
-            fid = Fid(ObjT.PROCESS.value, int(svc_id))
+            fid = str(Fid(ObjT.PROCESS.value, int(svc_id)))
             checks['args'].extend(['--fid', fid])
         elif name == 's3service':
-            fid = Fid(ObjT.PROCESS.value, int(svc_id))
-            s3svc = 's3server@' + str(fid)
+            fid = str(Fid(ObjT.PROCESS.value, int(svc_id)))
+            s3svc = 's3server@' + fid
```

Comment:
You can let fid be Fid object and convert it to str where it is used.
```
s3svc = 's3server@' + str(fid)
```
or have a helper function,
process_fid_str(svc_id: int):
    return str(Fid(ObjT.Process.value, svc_id)

Reply:
ok, lets keep it fid object

Reply:
done